pyaedt/modeler/cad/Modeler.py

```diff
@@ -1134,6 +1207,388 @@ def _attributes(self):
+        reverse_x_axis : bool, optional
+            Whether the x-axis is in the reverse direction.
+            The default is ``False``.
+        reverse_y_axis : bool, optional
+            Whether the y-axis is in the reverse direction.
+            The default is ``False``.
+        # y_axis_x_direction : float, optional
+        #     X direction of the y axis.
+        #     This option can be provided if the coordinate system type is either ``Rotated`` or ``Both``.
+        #     The default is ``0``.
+        # y_axis_y_direction : float, optional
+        #     Y direction of the y axis.
+        #     This option can be provided if the coordinate system type is either ``Rotated`` or ``Both``.
+        #     The default is ``0``.
+        # y_axis_z_direction : float, optional
+        #     Z direction of the y axis.
+        #     This option can be provided if the coordinate system type is either ``Rotated`` or ``Both``.
+        #     The default is ``0``.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        """
+        if cs_type == "Offset" or cs_type == "Both":
+            is_attached_to_entity = True
+            origin_entity_id = self._modeler.convert_to_selections(entity, True)[0]
+            if not isinstance(origin_entity_id, int):  # pragma: no cover
+                raise ValueError("Unable to find reference entity.")
+            else:
+                self.entity_id = origin_entity_id
+                o_type = self._get_type_from_id(self.entity_id)
+                if o_type == "Face":
+                    origin_position_type = "FaceCenter"
+                elif o_type == "Edge":
+                    origin_position_type = "EdgeCenter"
+                elif o_type == "Vertex":
+                    origin_position_type = "OnVertex"
+                else:  # pragma: no cover
+                    raise ValueError("origin must identify either Face or Edge or Vertex.")
+            if cs_type == "Both":
+                if not x_axis_entity:
+                    raise ValueError("Provide an entity to define the x axis direction.")
+                x_axis_entity_id = self._modeler.convert_to_selections(x_axis_entity, True)[0]
+                if not isinstance(x_axis_entity_id, int):  # pragma: no cover
+                    raise ValueError("Unable to find reference entity.")
+                else:
+                    o_type = self._get_type_from_id(x_axis_entity_id)
+                    if o_type == "Face":
+                        x_axis_position_type = "FaceCenter"
+                    elif o_type == "Edge":
+                        x_axis_position_type = "EdgeCenter"
+                    elif o_type == "Vertex":
+                        x_axis_position_type = "OnVertex"
+                    else:  # pragma: no cover
+                        raise ValueError("x axis must identify either Face or Edge or Vertex.")
```

Comment:
Same comment here for either tagging as possible values or lowercase to format as general text.